src/commands/dev.ts

```diff
@@ -353,9 +353,24 @@ export default class Dev extends BaseCommand {
       linked_components
     );
 
-    const port_available = await PortUtil.isPortAvailable(flags.port);
-    if (!port_available) {
-      this.error(`Could not run architect on port ${flags.port}.\nPlease stop an existing process or specify --port to choose a different port.`);
+    let port_available = await PortUtil.isPortAvailable(flags.port);
+    while (!port_available) {
```

Comment:
You could just turn this into
```
while (!(await PortUtil.isPortAvailable(flags.port)) {
```
then you don't need the variable at all and reduce the mutable state of the application.